packages/tailwindcss/src/index.test.ts

```diff
@@ -1292,6 +1292,68 @@ describe('Parsing themes values from CSS', () => {
       }"
     `)
   })
+
+  test('plugin theme values can override CSS theme values and behave as inline reference', async () => {
+    let { build } = await compile(
+      css`
+        @theme {
+          --color-red: red;
+          --color-orange: orange;
```

Comment:
Hm am I the only one who finds it confusing that we manually specify `--color-orange` here as a theme variable that is _not_ inline and yet, because of the plugin that we import later, it won't be emitted as CSS variable. I understand the reason that we don't want to emit CSS variables for plugins because of the weird names that some plugins use but if we overwrite something that is part of the user configured theme, I would kind of expect the CSS variable to still work (if someone writes a `@theme` like this, I think they expect `var(--color-orange)` to work. 🤔